regolith/tools.py

```diff
@@ -2102,6 +2105,176 @@ def google_cal_auth_flow():
+            (set up in user.json)
+        rc - run control object
+
+    Returns:
+        The token if the token exists and False if not
+    """
+
+    if token_info in rc:
+        if rc.__getattr__(token_info):
+            return rc.__getattr__(token_info)
+        else:
+            return False
+    else:
+        return False
+
+def create_repo(name, target_repo, token_info, rc):
+    """ Creates a repo if repo information and token is defined in rc
+
+    Parameters:
+        name - string
+            the name of the repo being created (e.g. 'test repo')
+        target_repo - string
+            the key in rc for target repo information (e.g. 'repo1')
+        token_info - string
+            the key in rc for token (e.g. 'priv_token')
+        rc - run control object
+
+    Returns:
+        Success message (repo target_repo has been created in talks) if repo is successfully created in target_repo
+        Warning/setup messages if unsuccessful (or if repo info or token are not valid)
+    """
+    repo_info_valid = repo_info_complete(target_repo, rc)
+    token_valid = token_info_complete(token_info, rc)
+    if repo_info_valid and token_valid:
+        try:
+            response = requests.post(rc.repos[target_repo]['url'], params=rc.repos[target_repo]['params'], headers={'PRIVATE-TOKEN': '{}'.format(token_valid)})
+            response.raise_for_status()
+            return f"repo {name} has been created in talks"
+        except requests.exceptions.HTTPError:
+            raise HTTPError(f"WARNING: Unsuccessful attempt at making a GitHub/GitLab etc., repository "
+                            f"due to an issue with the API call (status code: {response.status_code}). "
+                            f"Check that your private token and repository information are valid in "
+                            f"regolithrc.json.")
+        except requests.exceptions.RequestException as e:
+            raise SystemExit(e)
+    elif not repo_info_valid:
+        print("WARNING: The request URL info is not valid (parameters may not be defined). "
+              "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+              "please add repository information in regolithrc.json. See regolith documentation "
+              "for details.")
+    elif not token_valid:
+        print("WARNING: The authentication token may not be defined. If you would like regolith to "
+              "automatically create a repository in GitHub/GitLab, please add your private authentication "
+              "token in user.json. See regolith documentation for details.")
+    else:
+        print("If you would like regolith to automatically create a repository in GitHub/GitLab, "
+              "please add your repository information and private authentication token in reolgithrc.json"
+              "and user.json respectively. See regolith documentation for details.")
+
+def create_repo_old(name, repo_info, rc):
```

Comment:
delete this function so we can merge the PR.  save the code and paste it into a new branch where you will work on the url creation code.